__tests__/search.js

```diff
@@ -35,28 +35,51 @@ class MockVFSServiceProvider extends ServiceProvider {
     ];
   }
 }
+class MySearchAdapter {
+    constructor(core) {
+        this.core = core;
 
+    }
+
+    destroy() {
+    }
+
+    async init() {
+    }
+
+    async search(pattern) {
+        const list = [
+            {
+             path: 'jest:/test.jpg',
+             filename: pattern+'.jpg'   
```

Comment:
There's no need to use a pattern here. And the path and filename won't even match here.

Just do it like this:

```js
{
  path: 'custom:/test.jpg',
  filename: 'test.jpg',
}
```

And then update your test.